### DISCUSSION TITLE
Problems encountered during the migration to Vite: the 'entryFileNames' option in 'rollupOptions'.

### DISCUSSION THREAD

**Discussion by Mori-Yang** [farm-fe/farm#1546] [Q&A]
I want to migrate from Vite 4 to Farm, but I encountered an issue during the migration process: In Vite's configuration, I used rollupOptions.output.entryFileNames: (chunkInfo) => {}, which requires using a function to dynamically specify entryFileNames. However, in Farm, entryFileNames is a string. What should I do in this situation? Should I use partialBundling?

![image](https://github.com/farm-fe/farm/assets/145859385/8867d662-aad1-4175-ae8b-4311aff38296)

**Comment by wre232114** [comment]
farm does not support dynamic entry filenames for now. a plugin may be needed to achieve the same goal

**Reply by Mori-Yang** [reply]
Thanks for your answer, I will try to write the plugin！

**Reply by Mori-Yang** [reply]
Wondering if farm's communication groups (WeChat / QQ) are still accessible? How can I join it?

**Reply by fu050409** [reply]
> Wondering if farm's communication groups (WeChat / QQ) are still accessible? How can I join it?

They are available in `README.md`

**Reply by Mori-Yang** [reply]
Thank you all, I will discuss the subsequent migration issues in the group！！

**Reply by Mori-Yang** [reply]
Hi, I still can't solve this problem, there is no proper hook provided in the Farm plugin to handle the `entryfileName`, maybe I didn't notice it? I can use `options` Hook in Rollup, how about in Farm?

**Reply by wre232114** [reply]
Farm does not provide options to set entry filename dynamically. potential solutions:
1. write a js plugin that implement finalizeResources hook and update resources in this hook. But note the js plugin that uses this hook may be really slow
2. write a rust plugin that implement finalize_resources hook
